pkg/reconciler/cluster/cluster.go

```diff
@@ -80,18 +81,25 @@ func (c *Controller) reconcile(ctx context.Context, cluster *v1alpha1.Cluster) e
 	for resourceName, pulledCrd := range crds {
 		pulledCrd.SetClusterName(logicalCluster)
 		pulledCrd.Labels[clusterOriginLabel(cluster.Name)] = ""
-		clusterCrd, err := c.crdClient.CustomResourceDefinitions().Create(logicalClusterContext, pulledCrd, v1.CreateOptions{})
+		_, err := c.crdClient.CustomResourceDefinitions().Create(logicalClusterContext, pulledCrd, v1.CreateOptions{})
 		if k8serrors.IsAlreadyExists(err) {
+			var clusterCrd *apiextensionsv1.CustomResourceDefinition
 			clusterCrd, err = c.crdClient.CustomResourceDefinitions().Get(logicalClusterContext, pulledCrd.Name, v1.GetOptions{})
+
 			if err == nil {
 				if !equality.Semantic.DeepEqual(pulledCrd.Spec, clusterCrd.Spec) ||
-					!equality.Semantic.DeepEqual(pulledCrd.Annotations, clusterCrd.Annotations) ||
-					!equality.Semantic.DeepEqual(pulledCrd.Labels, clusterCrd.Labels) {
+					!equality.Semantic.DeepEqual(pulledCrd.Annotations, clusterCrd.Annotations) {
+
 					pulledCrd.ResourceVersion = clusterCrd.ResourceVersion
 					_, err = c.crdClient.CustomResourceDefinitions().Update(logicalClusterContext, pulledCrd, v1.UpdateOptions{})
 					if k8serrors.IsConflict(err) {
 						return errors.NewRetryableError(fmt.Errorf("Conflict when applying CRD pulled from cluster %s for resource %s: %v", cluster.Name, resourceName, err))
 					}
+				} else if !equality.Semantic.DeepEqual(pulledCrd.Labels, clusterCrd.Labels) {
+					if _, ok := clusterCrd.Labels[clusterOriginLabel(cluster.Name)]; !ok {
+						clusterCrd.Labels[clusterOriginLabel(cluster.Name)] = ""
+						_, err = c.crdClient.CustomResourceDefinitions().Update(logicalClusterContext, clusterCrd, v1.UpdateOptions{})
```

Comment:
Please could you add the same check in the second `Update` as the check done in [line 95](https://github.com/kcp-dev/kcp/pull/88/files#diff-6bda1f49bbe6d47cbfdff2057c2e38ea3d12129b78e9c19837b7359f4bf0d06fR95) ?

Reply:
Done